ui/src/app/routing/signed-in-app-routing.tsx

```diff
@@ -138,8 +139,18 @@ const WorkspaceSearchAdminPage = fp.flow(
 )(AdminWorkspaceSearch);
 
 export const SignedInRoutes = () => {
-  const { search } = useLocation();
-  const query = new URLSearchParams(search);
+  const location = useLocation();
+  const query = new URLSearchParams(location.search);
+  const [pathName, setPathName] = React.useState(location.pathname);
+  React.useEffect(() => {
+    if (
+      pathName.startsWith('/workspaces/') &&
+      !location.pathname.startsWith('/workspaces/')
+    ) {
+      currentWorkspaceStore.next(null);
```

Comment:
To test this, set your credits to expired, and navigate to your workspace and then navigate to your profile page.

To ensure that there was a problem before, comment out this line and try the same navigation.